lib/friendly_shipping/services/ups/serialize_shipment_confirm_request.rb

```diff
@@ -0,0 +1,274 @@
+# frozen_string_literal: true
+
+require 'friendly_shipping/services/ups/serialize_shipment_address_snippet'
+require 'friendly_shipping/services/ups/serialize_package_node'
+
+module FriendlyShipping
+  module Services
+    class Ups
+      class SerializeShipmentConfirmRequest
+        class << self
+          # Item options (only necessary for international shipping)
+          #
+          # @option description [String] A description of the item for the intl. invoice
+          # @option commodity_code [String] Commodity code for the item. See https://www.tariffnumber.com/
+          # @option value [Money] Price of the item
+          # @option unit_of_item_count [String] What kind of thing is one of this item? Example: Barrel, m3.
+          #   See UPS docs for codes.
+          #
+          def call(
+            shipment:,
+            options:
+          )
+            xml_builder = Nokogiri::XML::Builder.new do |xml|
+              xml.ShipmentConfirmRequest do
+                xml.Request do
+                  xml.RequestAction('ShipConfirm')
+                  # Required element controls level of address validation.
+                  xml.RequestOption(options.validate_address ? 'validate' : 'nonvalidate')
+                  xml.SubVersion('1707')
+                  # Optional element to identify transactions between client and server.
+                  if options.customer_context
+                    xml.TransactionReference do
+                      xml.CustomerContext(options.customer_context)
+                    end
+                  end
+                end
+
+                xml.Shipment do
+                  xml.Service do
+                    xml.Code(options.shipping_method.service_code)
+                  end
+
+                  xml.ShipTo do
+                    SerializeShipmentAddressSnippet.call(xml: xml, location: shipment.destination)
+                  end
+
+                  # Required element. The company whose account is responsible for the label(s).
+                  xml.Shipper do
+                    SerializeShipmentAddressSnippet.call(xml: xml, location: options.shipper || shipment.origin)
+
+                    xml.ShipperNumber(options.shipper_number)
+                  end
+
+                  if options.shipper
+                    xml.ShipFrom do
+                      SerializeShipmentAddressSnippet.call(xml: xml, location: shipment.origin)
+                    end
+                  end
+
+                  if options.negotiated_rates
+                    xml.RateInformation do
+                      xml.NegotiatedRatesIndicator
+                    end
+                  end
+
+                  # shipment.options.fetch(:reference_numbers, {}).each do |reference_code, reference_value|
+                  #   xml.ReferenceNumber do
+                  #     xml.Code(reference_code)
+                  #     xml.Value(reference_value)
+                  #   end
+                  # end
```

Comment:
Does this need to be left here for some reason?

Reply:
Good call. Yes, we'll need to think about how to properly do this, that's why it's still in here. For some international shipments, package-level reference numbers are not allowed, and instead, shipment-level reference numbers are allowed. Question: Should we support? :)